### ISSUE TITLE
BT-36 : duration period, days

### ISSUE THREAD

**Issue by MiraRatajczak** [eForms/eForms#83]
Why is the estimation only expressed in days? In the current forms we also have the possibility to express in months. Are these calendar days or working days?

**Comment by JachymHercher** [comment]
Because any other unit (e.g. month, year) can be converted to days (or back) in the eTendering system / displaying system. We don't see a reason to store data in multiple formats in the standard itself. 

Calendar days, thank you for pointing it out. (Calendar days are better than working days because working days are different from country to country, and thus their use could lead to misunderstandings. Thanks to @guascce for pointing this out.)  

We suggest changing the definition of BT-36 to:

"The (estimated) number of _calendar_ days from the start to the end of the contract, framework agreement, dynamic purchasing system or qualification system. This must include any options and renewals."

**Comment by tinne** [comment]
Or can they? As far as I remember three months comprise of between 89 and 92 days, which can make a difference, especially when exact time periods are hard to determine in advance (cf. #98).

**Comment by JachymHercher** [comment]
Yes, there can be a few days differences. However, as long as this is clear on the input side (i.e. in the eTendring system), I'm not sure whether it poses a problem in practice. (E.g. I doubt many buyers intentionally use the flexibility of there being different numbers of days in a month.)

**Comment by tinne** [comment]
Talking XML schema, there is the ”duration“ data type, which allows a duration of time expressed as a number of years, months, days (and inter-day durations, which can be lexically forbidden). Using such a definition of a time interval might feel more intuitive for the buyers.

In input and print, this may boil down to expressions like „15 days,“ „6 months,“ or „2 years and 6 months.“

**Comment by jpmckinney** [comment]
I believe @myroslav had a use case in Prozorro for expressing periods as a number of months rather than a number of days.

**Comment by myroslav** [comment]
In ProZorro.sale we already use [ISO 8601 Durations](https://en.wikipedia.org/wiki/ISO_8601#Durations). See http://lotsloki.api-docs.registry.ea2.openprocurement.io/en/latest/standard/auction.html#schema for `tenderingDuration` field and http://lotsloki.api-docs.registry.ea2.openprocurement.io/en/latest/tutorial.html#creating-lot for the field use patterns. We will be using [ISO 8601 Durations](https://en.wikipedia.org/wiki/ISO_8601#Durations) in ProZorro Framework Agreements as well (when describing duration of Agreement to be signed).

[ISO 8601 Durations](https://en.wikipedia.org/wiki/ISO_8601#Durations) are quite verbose and does allow periods that are measured in years and months (i.e. variable days), that is quite usable in legal environments where longer durations are expressed in such units.

For instance

 * `"P4Y"` is 4 years,
 * `"P3M"` is 3 months and 
 * `"P21D"` is 21 days, but it is also possible to express the same as `"P3W"`.

The format is very usable for machine interpretation, but not too usable in Excel/Google Spreadsheets, until ISO duration parser is implemented in VB/Google Apps Script.

Note that [schema.org durations](https://schema.org/duration) are also ISO8601 (be aware that 1 out of 3 examples has an error in ISO duration format, no `P` prefix).

**Comment by JachymHercher** [comment]
Thanks, everyone. We will come back to the technical side of this discussion in a few months when we are working on the technical implementation. 

What I take from it for now is that we shouldn't be too prescriptive in the implementing regulation. We will change the definition to reflect that, probably by simply replacing "number of days" to "duration".

**Comment by tinne** [comment]
So, suggest the following definition for BT-36
> The (estimated) duration between of the contract, framework agreement, dynamic purchasing system or qualification system. This must include any options and renewals.

Data type Duration could be described as

> A set of information that defines, for any given calendar date ”start,“ a valid calendar date ”end.“

**Comment by tinne** [comment]
@JachymHercher Sorry to confuse you, can you elaborate on that?

**Comment by JachymHercher** [comment]
The confusion just referred to a typo - "The (estimated) duration between of the contract".

In my previous message, the definition I had in mind was: 

> The (estimated) duration from the start to the end of the contract, framework agreement, dynamic purchasing system or qualification system. This must include any options and renewals.

**Comment by tinne** [comment]
Oops, yes, now updated. Guess moving the fact that a time frame must have a beginning and an end to the data type definition might only marginally lift consiseness.

**Comment by JachymHercher** [comment]
Yes, we originally had it that way, but then decided to leave it in the BT definition, because there weren't many fields with the "Duration" data type.
